packages/shared/src/lib/constants.ts

```diff
@@ -15,3 +18,16 @@ export const isDevelopment = process.env.NODE_ENV === 'development';
 export const isProduction = process.env.NODE_ENV === 'production';
 export const isTesting =
   process.env.NODE_ENV === 'test' || (!isDevelopment && !isProduction);
+
+export const isBrave = (): boolean => {
+  if (!window.Promise) {
+    return false;
+  }
+  if (
+    typeof navigator.brave === 'undefined' ||
+    typeof navigator.brave.isBrave !== 'function'
+  ) {
+    return false;
+  }
+  return true;
```

Comment:
I think utilizing the optional chaining and returning the result of the statement is the way to go :)

```suggestion
  return typeof navigator.brave?.isBrave === 'function';
```